layers/+lang/typescript/packages.el

```diff
@@ -14,22 +14,43 @@
         company
         eldoc
         flycheck
+        rainbow-identifiers
         tide
         typescript-mode
         web-mode
         ))
 
+(defmacro typescript||iterate-modes (&rest body)
+  `(progn
+     ,@(loop for mode in (if typescript-use-web-mode-for-ts '(web-typescript-mode) '(typescript-mode web-typescript-mode))
+             collect (let ((hook (intern (format "%S-hook" mode)))
+                           (jump-handlers (intern (concat "spacemacs-jump-handlers-" (symbol-name mode)))))
+                       `(progn
+                          ,@(subst mode 'mode (subst hook 'hook (subst jump-handlers 'jump-handlers body))))))))
+
 (defun typescript/post-init-company ()
   (when (configuration-layer/package-usedp 'tide)
-    (spacemacs|add-company-backends
-      :backends company-tide
-      :modes typescript-mode)))
+    (typescript||iterate-modes
+     (spacemacs|add-company-backends
+       :backends company-tide
+       :modes mode))))
+
+(defun typescript/post-init-rainbow-identifiers ()
+  (add-to-list 'rainbow-identifiers-faces-to-override 'web-mode-variable-name-face)
+  (typescript||iterate-modes
+   (add-hook 'hook 'rainbow-identifiers-mode)))
 
 (defun typescript/post-init-eldoc ()
-  (add-hook 'typescript-mode-hook 'eldoc-mode))
+  (typescript||iterate-modes
+   (add-hook 'hook 'eldoc-mode)))
 
 (defun typescript/post-init-flycheck ()
-  (spacemacs/enable-flycheck 'typescript-mode))
+  (spacemacs|use-package-add-hook tide
+    :post-config
+    (typescript||iterate-modes
+     (flycheck-add-mode 'typescript-tide 'mode)
+     (flycheck-add-mode 'typescript-tslint 'mode)
```

Comment:
I haven't tested it yet, but this might also run too late to get added to the first file. Is there a reason this got added to tide's post-config hook?